lib/appmap/hook/method.rb

```diff
@@ -45,42 +47,48 @@ def activate
+        hook_method_def = Proc.new do |*args, &block|
+          instance_method = hook_method.bind(self).to_proc
+          call_instance_method = -> {
+            # https://github.com/applandinc/appmap-ruby/issues/153
+            if Util.ruby_minor_version >= 2.7 && args == ARRAY_OF_EMPTY_HASH && hook_method.arity == 1
+              instance_method.call({}, &block)
+            else
+              instance_method.call(*args, &block)
+            end
+          }
 
-            # We may not have gotten the class for the method during
-            # initialization (e.g. for a singleton method on an embedded
-            # struct), so make sure we have it now.
-            defined_class, = Hook.qualify_method_name(hook_method) unless defined_class
+          # We may not have gotten the class for the method during
+          # initialization (e.g. for a singleton method on an embedded
+          # struct), so make sure we have it now.
+          defined_class, = Hook.qualify_method_name(hook_method) unless defined_class
 
-            reentrant = Thread.current[HOOK_DISABLE_KEY]
-            disabled_by_shallow_flag = \
-              -> { hook_package&.shallow? && AppMap.tracing.last_package_for_current_thread == hook_package }
+          reentrant = Thread.current[HOOK_DISABLE_KEY]
+          disabled_by_shallow_flag = \
+            -> { hook_package&.shallow? && AppMap.tracing.last_package_for_current_thread == hook_package }
 
-            enabled = true if AppMap.tracing.enabled? && !reentrant && !disabled_by_shallow_flag.call
+          enabled = true if AppMap.tracing.enabled? && !reentrant && !disabled_by_shallow_flag.call
 
-            return call_instance_method.call unless enabled
+          return call_instance_method.call unless enabled
 
-            call_event, start_time = with_disabled_hook.call do
-              before_hook.call(self, defined_class, args)
-            end
-            return_value = nil
-            exception = nil
-            begin
-              return_value = call_instance_method.call
-            rescue
-              exception = $ERROR_INFO
-              raise
-            ensure
-              with_disabled_hook.call do
-                after_hook.call(self, call_event, start_time, return_value, exception) if call_event
-              end
+          call_event, start_time = with_disabled_hook.call do
+            before_hook.call(self, defined_class, args)
+          end
+          return_value = nil
+          exception = nil
+          begin
+            return_value = call_instance_method.call
+          rescue
+            exception = $ERROR_INFO
+            raise
+          ensure
+            with_disabled_hook.call do
+              after_hook.call(self, call_event, start_time, return_value, exception) if call_event
             end
           end
         end
+        hook_method_def = hook_method_def.ruby2_keywords if Util.ruby_minor_version >= 2.7
```

Comment:
This is probably how the problem above is introduced - but it's necessary to have the proper behavior in Ruby 2.7.

Reply:
The doc suggests using `respond_to?(:ruby2_keywords)`. Is there any reason to prefer that?

Reply:
So, you're saying this causes a problem, and the change above works around the problem, right? Can you give an example of the bad behavior this change fixes, please?

Reply:
So, without calling `ruby2_keywords`, then in the case above, args is `[]` and not `[{}]`, so there's really no way to know what the argument to the proxied method should be. At least with `ruby2_keywords`, it's possible to determine what should be happening.

Reply:
We can use `respond_to?`